tools/checked-c-convert/ConstraintBuilder.cpp

```diff
@@ -268,27 +268,32 @@ class FunctionVisitor : public RecursiveASTVisitor<FunctionVisitor> {
             // to a NamedDecl?
             FunctionDecl *calleeDecl =
               dyn_cast<FunctionDecl>(CA->getCalleeDecl());
-            if (calleeDecl && calleeDecl->getName() == "malloc") {
-              // It's a call to malloc. What about the parameter to the call?
-              if (CA->getNumArgs() > 0) {
-                UnaryExprOrTypeTraitExpr *arg =
-                  dyn_cast<UnaryExprOrTypeTraitExpr>(CA->getArg(0));
-                if (arg && arg->isArgumentType()) {
-                  // Check that the argument is a sizeof.
-                  if (arg->getKind() == UETT_SizeOf) {
-                    QualType argTy = arg->getArgumentType();
-                    // argTy should be made a pointer, then compared for
-                    // equality to lhsType and rhsTy.
-                    QualType argPTy = Context->getPointerType(argTy);
-
-                    if (Info.checkStructuralEquality(V, RHSConstraints, argPTy, lhsType) &&
-                        Info.checkStructuralEquality(V, RHSConstraints, argPTy, rhsTy)) {
-                      rulesFired = true;
-                      // At present, I don't think we need to add an
-                      // implication based constraint since this rule only
-                      // fires if there is a cast from a call to malloc.
-                      // Since malloc is an external, there's no point in
-                      // adding constraints to it.
+            if (calleeDecl && isFunctionAllocator(calleeDecl->getName())) {
+              // this is an allocator, should we treat it as safe?
+              if (!considerAllocUnsafe) {
+                rulesFired = true;
+              } else {
+                // It's a call to allocator. What about the parameter to the call?
+                if (CA->getNumArgs() > 0) {
```

Comment:
This code isn't correct for `calloc` or `realloc`.  For `calloc` and `realloc`, sizeof would be the second argument.

Reply:
The code is specifically for malloc. Fixed it.